### PR TITLE
Remove circular_file.cc from codebase, cmake, qa, and ConfigChecks #2874

### LINKED ISSUE #2874: runtime: remove unused circular_file.cc
* and all references to it in CMake
* and all qa_circular_file
* and the header checks from gnuradio-runtime/ConfigChecks.cmake `GR_CHECK_HDR_N_DEF(io.h HAVE_IO_H)` isn't necessary afterwards

### PR DESCRIPTION
**Pull request by grant-cox**
Removing unused files, qa, and cmake checks. Closes #2874 . I accidentally created a draft merge request for this previously and closed it (sorry for that). Let me know about any changes needed, thanks.

### CODE REVIEW

**Comment by michaelld** [comment]
Nice to get rid of unused cruft!

File: gr-blocks/lib/ConfigChecks.cmake [hunk 1]
```diff
@@ -28,7 +28,6 @@ ENDIF(MSVC)
 GR_CHECK_HDR_N_DEF(sys/time.h HAVE_SYS_TIME_H)
 GR_CHECK_HDR_N_DEF(sys/types.h HAVE_SYS_TYPES_H)
 GR_CHECK_HDR_N_DEF(sys/socket.h HAVE_SYS_SOCKET_H)
-GR_CHECK_HDR_N_DEF(io.h HAVE_IO_H)
```
Comment (marcusmueller) [inline gr-blocks/lib/ConfigChecks.cmake]: Mustn't be removed – HAVE_IO_H is still used in gr-blocks!

**Comment by marcusmueller** [comment]
Hey, this is so cool! Thanks!

So, sorry for not reacting earlier in #2874: in gr-blocks, the check for io.h still is necessary, since that is used there!

**Comment by mbr0wn** [comment]
Just adding the CLA label here because I'm not sure if @grant-cox already has a CLA executed. @grant-cox -- do you? If so, sorry for holding this up, if not, please contact bhilburn at gnuradio dot org to get that process started -- it's pretty painless!

**Comment by ValZapod** [comment]
Can you do a cla bot? )) like here https://github.com/microsoft/terminal/pull/3228
It is everywhere around github, you know...

**Comment by michaelld** [comment]
> Can you do a cla bot? )) like here [microsoft/terminal#3228](https://github.com/microsoft/terminal/pull/3228)
> It is everywhere around github, you know...

@bhilburn is the on who handles CLAs ... thinking we can't do a CLA bot ...

**Comment by ValZapod** [comment]
Are you sure)) FUI, then. https://github.com/apps/cla-bot

**Comment by michaelld** [comment]
> Are you sure)) FUI, then. https://github.com/apps/cla-bot

It's not up to me. For CLAs @bhilburn is who handles them. It's up to him & upstream (the FSF) to determine how they are handled.

**Comment by grant-cox** [comment]
@mbr0wn @michaelld @marcusmueller I pushed a commit to add the cmake checks back. I will contact bhilburn about filling out the CLA. Thanks for the help!

**Comment by marcusmueller** [comment]
You don't need a CLA for code removal, just for >10 LOC additions.

**Comment by marcusmueller** [comment]
@ValZapod we could do a CLA bot, but our CLAs go through the FSF and we don't have a complete list of a signees. Also, please don't raise such issues in a foreign pull request, but on the mailing list!

**Pull request merged**